scripts/helpers/wallet-precalculation.helper.js

```diff
@@ -0,0 +1,243 @@
+import { Address, Script } from 'bitcore-lib';
+import config from '../../__tests__/integration/configuration/config-fixture';
+
+export const precalculationHelpers = {
+  /**
+   * @type WalletPrecalculationHelper
+   */
+  test: null
+};
+
+export const multisigWalletsData = {
+  words: [
+    'upon tennis increase embark dismiss diamond monitor face magnet jungle scout salute rural master shoulder cry juice jeans radar present close meat antenna mind',
+    'sample garment fun depart various renew require surge service undo cinnamon squeeze hundred nasty gasp ridge surge defense relax turtle wet antique october occur',
+    'intact wool rigid diary mountain issue tiny ugly swing rib alone base fold satoshi drift poverty autumn mansion state globe plug ancient pudding hope',
+    'monster opinion bracket aspect mask labor obvious hat matrix exact canoe race shift episode plastic debris dash sort motion juice leg mushroom maximum evidence',
+    'tilt lab swear uncle prize favorite river myth assault transfer venue soap lady someone marine reject fork brain swallow notice glad salt sudden pottery',
+  ],
+  pubkeys: [
+    'xpub6CvvCBtHqFfErbcW2Rv28TmZ3MqcFuWQVKGg8xDzLeAwEAHRz9LBTgSFSj7B99scSvZGbq6TxAyyATA9b6cnwsgduNs9NGKQJnEQr3PYtwK',
+    'xpub6CA16g2qPwukWAWBMdJKU3p2fQEEi831W3WAs2nesuCzPhbrG29aJsoRDSEDT4Ac3smqSk51uuv6oujU3MAAL3d1Nm87q9GDwE3HRGQLjdP',
+    'xpub6BwNT613Vzy7ARVHDEpoX23SMBEZQMJXdqTWYjQKvJZJVDBjEemU38exJEhc6qbFVc4MmarN68gUKHkyZ3NEgXXCbWtoXXGouHpwMEcXJLf',
+    'xpub6DCyPHg4AwXsdiMh7QSTHR7afmNVwZKHBBMFUiy5aCYQNaWp68ceQXYXCGQr5fZyLAe5hiJDdXrq6w3AXzvVmjFX9F7EdM87repxJEhsmjL',
+    'xpub6CgPUcCCJ9pAK7Rj52hwkxTutSRv91Fq74Hx1SjN62eg6Mp3S3YCJFPChPaDjpp9jCbCZHibBgdKnfNdq6hE9umyjyZKUCySBNF7wkoG4uK',
+  ],
+  walletConfig: {
+    pubkeys: [],
+    total: 5,
+    minSignatures: 3,
+  }
+};
+multisigWalletsData.walletConfig.pubkeys = multisigWalletsData.pubkeys;
+
+export class WalletPrecalculationHelper {
+  WALLETS_FILENAME = '';
+
+  walletsDb = [];
+
+  /**
+   * Initializes the helper with a filename to sync the local wallet storage with.
+   * @param walletsFilename
+   */
+  constructor(walletsFilename) {
+    this.WALLETS_FILENAME = walletsFilename || './tmp/wallets.txt';
+  }
+
+  /**
+   * Generates 22 addresses for a wallet 24-word seed.
+   * @param [params]
+   * @param {string} [params.words] Optional wallet seed words. If empty, generates a new wallet
+   * @param {number} [params.addressIntervalStart=0] Optional interval start index
+   * @param {number} [params.addressIntervalEnd=22] Optional interval end index
+   * @param {{minSignatures:number,wordsArray:string[]}} [params.multisig] Optional multisig object
+   * @returns {{addresses: string[], words: string}}
+   */
+  static generateAddressesForWordsSeed(params = {}) {
+    const timeStart = Date.now().valueOf();
+
+    // Generating seed if none was informed
+    let wordsInput = params.words;
+    if (!wordsInput) {
```

Comment:
This should be inside the `else` below because it makes sense to generate new words only if it's not multisig.

Reply:
Good point. Moved that code on [3c6e4fc](https://github.com/HathorNetwork/hathor-wallet-headless/pull/189/commits/3c6e4fc97e977eaf1e5a571d25742a19b67414c0) and updated the JSDocs as well.